src/test/java/com/vinyl/AddProductToCartTest.java

```diff
@@ -0,0 +1,132 @@
+        ResponseEntity<?> response = trt.exchange("/products/" + product.getId()+ "/cart",HttpMethod.POST,new HttpEntity<>(request,headers),Void.class);
+
+        return response;
+    }
+
+    @Test
+    public void testWhenUserLoggedInAndItemExistsInCart(){
+        Token token = defaultEntitiesHelper.createToken(user);
+        Cart cart =defaultEntitiesHelper.createCart(user);
+        ProductCart pc=defaultEntitiesHelper.createProductCart(cart, product);
+        cart.setProducts(Lists.newArrayList(pc));
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse(token);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+        Assertions.assertThat(productCartRepository.findByProductAndCart(product,cart).getNrItems()).isEqualTo(pc.getNrItems() + request.getQuantity());
+    }
+
+    @Test
+    public void testWhenUserLoggedInAndItemDoesNotExistInCart() {
+        Token token = defaultEntitiesHelper.createToken(user);
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse(token);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.CREATED);
+        Cart cart = cartRepository.findByUser(user);
+        Assertions.assertThat(productCartRepository.findByProductAndCart(product, cart).getNrItems()).isEqualTo(request.getQuantity());
+    }
+
+    @Test
+    public void testWhenTokenIsMissing(){
+        ResponseEntity<?> response = trt.exchange("/products/" + product.getId()+ "/cart",HttpMethod.POST,new HttpEntity<>(request),Void.class);
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.UNAUTHORIZED);
+    }
+
+    @Test
+    public void testWhenTokenIsInvalid() {
+        HttpHeaders headers=tokenHeaderHelper.setupToken("INVALID_TOKEN");
+        ResponseEntity<?> response = trt.exchange("/products/" + product.getId()+ "/cart",HttpMethod.POST,new HttpEntity<>(request,headers),Void.class);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenTokenIsExpired() {
+        Token token = defaultEntitiesHelper.createToken(user);
+        token.setValidUntil(LocalDate.now().minusMonths(3));
+        tokenRepository.save(token);
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse(token);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.UNAUTHORIZED);
+    }
+
+    @Test
+    public void testWhenQuantityIsTooLarge() {
+        Token token = defaultEntitiesHelper.createToken(user);
+        product.setStock(900);
+        request.setQuantity(100000);
+
+        ResponseEntity<?> response = setUpHeaderAndGetTheResponse(token);
+
+        Assertions.assertThat(response.getStatusCode()).isEqualTo(HttpStatus.BAD_REQUEST);
+    }
+
+    @Test
+    public void testWhenProductIsInvalid() {
+        Token token = defaultEntitiesHelper.createToken(user);
+        product.setId(0000);
```

Comment:
here you shouldn't set an invalid id to the product field, you should do a trt.exchange with an invalid product id

Reply:
fixed